src/components/token-manager.js

```diff
@@ -45,28 +60,53 @@ class TokenManager {
           backgroundColor: _class && _class.color ? _class.color : null,
         });
         selectedTokens = [];
-        newTokens.push(t);
+        newTokens.push(currentToken);
       } else {
-        newTokens.push(t);
+        newTokens.push(currentToken);
       }
     }
 
     // Case if the selected tokens are at the end of the text and have not been added to the newTokens
     if (selectedTokens.length) {
-        newTokens.push({
-          type: "token-block",
-          start: selectedTokens[0].start,
-          end: selectedTokens[selectedTokens.length - 1].end,
-          tokens: selectedTokens,
-          label: _class && _class.name ? _class.name : "Unlabelled",
-          classId: _class && _class.id ? _class.id : 0,
-          backgroundColor: _class && _class.color ? _class.color : null,
+      newTokens.push({
+        type: "token-block",
+        start: selectedTokens[0].start,
+        end: selectedTokens[selectedTokens.length - 1].end,
+        tokens: selectedTokens,
+        label: _class && _class.name ? _class.name : "Unlabelled",
+        classId: _class && _class.id ? _class.id : 0,
+        backgroundColor: _class && _class.color ? _class.color : null,
       });
     }
 
     this.tokens = newTokens;
   }
 
+  /**
+   * 
+   */
+  selectNewBlock(selection, _class) {
+    if (
+      selection.anchorOffset === selection.focusOffset &&
+      selection.anchorNode === selection.focusNode
+    ) {
+      return;
+    }
+    
+    const range = selection.getRangeAt(0);
+    let start, end;
+    try {
+      start = parseInt(range.startContainer.parentElement.id.replace("t", ""));
+      let offsetEnd = parseInt(range.endContainer.parentElement.id.replace("t", ""));
+      end = offsetEnd + range.endOffset;
```

Comment:
It would be better to keep the DOM related stuff in the Vue code and keep the token manager as vanilla JS as possible. Also this alert seems to have been removed, which creates "Unlabelled" tags instead of alerting the user that they haven't created any tags.

```js
      if (!this.classes.length && selection.anchorNode) {
        alert(
          "There are no Tags available. Kindly add some Tags before tagging."
        );
        selection.empty();
        return;
      }
```

Reply:
Moving stuff back in [218d98a](https://github.com/tecoholic/ner-annotator/pull/47/commits/218d98afffd8fa096c409fcfc63092f8feae5d26)